addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -29,8 +29,68 @@ type ManagedClusterAddOn struct {
 	Status ManagedClusterAddOnStatus `json:"status"`
 }
 
-// ManagedClusterAddOnSpec is empty for now.
+// ManagedClusterAddOnSpec defines the install configuration of
+// an addon agent on managed cluster.
 type ManagedClusterAddOnSpec struct {
+	// installNamespace is the namespace on the managed cluster to install the addon agent.
+	// If it is not set, open-cluster-management-addon namespace is used to install the addon agent.
+	// +optional
+	InstallNamespace string `json:"installNamespace,omitempty"`
+
+	// leaseDurationSeconds is used to coordinate the lease update time of addon agents on the managed cluster.
+	// The addon agent should maintain a lease with the name of addon name in the namespace of installNamespace.
+	// The Klusterlet agent will monitor the lease and update the status of the related managedclusteraddon
+	// resource accordingly.  If its value is zero, the addon agent should update its lease every 60 seconds by default
+	// +optional
+	LeaseDurationSeconds int32 `json:"leaseDurationSeconds,omitempty"`
```

Comment:
This is slightly less clear to me.  do you mind making the case the in a separate PR.  Is your goal here to allow longer times or shorter times?  Who would choose to change this and why?

Reply:
removed for now. You are right, I have to rethink on this.